metric_providers/cpu/energy/rapl/msr/component/Makefile

```diff
@@ -1,6 +1,6 @@
 CFLAGS = -O3 -Wall -Werror -lm -I../../../../../../lib/c
 
-metric-provider-binary: source.c
+metric-provider-binary: source.c ../../../../../../lib/c/gmt-lib.o
```

Comment:
To my understanding this  will track the change and rebuild metric-provider-binary if the .o file has changed. But it will also try to rebuild the o-file if  their source has changed, not? I wonder if you can build the source from here ... are all the headers / includes set?

Testing would be that you only change the source of the gmt-lib.c and see if the make of the metric-provider-binary rebuilds fully.

Reply:
Correct, the rebuild happens if the source of the dependency has changed. I tested it with gmt-lib.c: If I add a comment, a rebuild happens. However, a change in the header file does not lead to a rebuild. If that should be the case, the header file has to be added to it too as a dependency:

```sh
metric-provider-binary: source.c ../../../../../lib/c/gmt-lib.o ../../../../../lib/c/gmt-lib.h
```

Should I do that?

Reply:
hmm actually unsure. I have never seen header files as dependencies. @ribalba Do you have insights here?

Reply:
I would include the header files too. What we want is that if the functionality would change you should recompile which can happen if you change the `.h` file. 

Let's say you change some buffer size in the `.h` file and then want to test the metric provider you would expect it to have the new buffer size.

Reply:
Done. I added the header files as dependencies. I also replaced the long paths to the lib/c directory with a variable.